buildbot/compile.py

```diff
@@ -12,7 +12,7 @@ def do_compile(args):
     if cpu_count is None:
         cpu_count = DEFAULT_CPU_COUNT
 
-    make_cmd = ["ninja", "-j", str(cpu_count), "sycl-toolchain"]
+    make_cmd = ["ninja", "-j", str(cpu_count), "sycl-toolchain", "SYCLUnitTests"]
```

Comment:
This ensures that all SYCL unittests are build, too.

Reply:
Actually it must be done at "check" step.
Usually the tests are built as a dependency for the "check-*" targets.

Reply:
With this I try to ensure that all building happens in the "compile" phase so the "check" phase only runs testing. In a Jenkins test set up this would lead to a clearer separation of stages to 
1. build once, 
2. store the build artifacts (the SYCL compiler), and then 
3. during a test stage only use the generated binaries (including the SYCL compiler).

@bader Would you like me to remove the change?

Reply:
@bjoernknafla, does it work well for LIT tests?
I thought that some configuration by cmake is required + test sources.

If this not a problem, we can get list of targets to build from environment variable set in Jenkins/Buildbot.
If env. variable is not set, we suggest using default list of targets w/o tests.
Does it make sense to you?

Reply:
Thanks to your questions I have figured out how Clang makes its unittests part of the LIT related `check` target. When building the LIT tests the unittests are now also build. No need to add `SYCLUnitTests` to the build script.